test/test_transforms_tensor.py

```diff
@@ -715,3 +718,78 @@ def test_gaussian_blur(device, meth_kwargs):
     )
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('alphas', [
+    {"mixup_alpha": 1.0, "cutmix_alpha": 1.0, 'cutmix_p': 1.0},
+    {"mixup_alpha": 1.0, "cutmix_alpha": 1.0, 'cutmix_p': 0.0},
+    {"mixup_alpha": 1.0, "cutmix_alpha": 1.0, 'p': 0.0},
+    {"mixup_alpha": 0.0, "cutmix_alpha": 1.0},
+    {"mixup_alpha": 1.0, "cutmix_alpha": 0.0},
+])
+@pytest.mark.parametrize('label_smoothing', [0.0, 0.1])
+@pytest.mark.parametrize('inplace', [True, False])
+def test_random_mixupcutmix(device, alphas, label_smoothing, inplace):
+    batch_size = 32
+    num_classes = 10
+    batch = torch.rand(batch_size, 3, 44, 56, device=device)
+    targets = torch.randint(num_classes, (batch_size, ), device=device, dtype=torch.int64)
+
+    fn = T.RandomMixupCutmix(num_classes, label_smoothing=label_smoothing, inplace=inplace, **alphas)
+    scripted_fn = torch.jit.script(fn)
+
+    seed = torch.seed()
+    output = fn(batch.clone(), targets.clone())
+
+    torch.manual_seed(seed)
+    output_scripted = scripted_fn(batch.clone(), targets.clone())
+    assert_equal(output[0], output_scripted[0])
+    assert_equal(output[1], output_scripted[1])
+
+    fn.__repr__()
+
+
+def test_random_mixupcutmix_with_invalid_data():
+    with pytest.raises(AssertionError, match="Please provide a valid positive value for the num_classes."):
+        T.RandomMixupCutmix(0)
+    with pytest.raises(AssertionError, match="Both alpha params can't be zero."):
+        T.RandomMixupCutmix(10, mixup_alpha=0.0, cutmix_alpha=0.0)
+
+    t = T.RandomMixupCutmix(10)
+    with pytest.raises(ValueError, match="Batch ndim should be 4."):
+        t(torch.rand(3, 60, 60), torch.randint(10, (1, )))
+    with pytest.raises(ValueError, match="Target ndim should be 1."):
+        t(torch.rand(32, 3, 60, 60), torch.randint(10, (32, 1)))
+    with pytest.raises(ValueError, match="Target dtype should be torch.int64."):
+        t(torch.rand(32, 3, 60, 60), torch.randint(10, (32, ), dtype=torch.int32))
+    with pytest.raises(ValueError, match="The batch size should be even."):
+        t(torch.rand(31, 3, 60, 60), torch.randint(10, (31, )))
+
+
+def test_random_mixupcutmix_with_real_data():
+    torch.manual_seed(12)
+
+    # Build dummy dataset
+    images = []
+    for test_file in [("encode_jpeg", "grace_hopper_517x606.jpg"), ("fakedata", "logos", "rgb_pytorch.png")]:
+        fullpath = (os.path.dirname(os.path.abspath(__file__)), 'assets') + test_file
+        img = read_image(get_file_path_2(*fullpath))
+        images.append(F.resize(img, [224, 224]))
+    dataset = TensorDataset(torch.stack(images).to(torch.float32), torch.tensor([0, 1]))
+
+    # Use mixup in the collate
+    mixup = T.RandomMixupCutmix(2, cutmix_alpha=1.0, mixup_alpha=1.0, label_smoothing=0.1)
+    dataloader = DataLoader(dataset, batch_size=2,
+                            collate_fn=lambda batch: mixup(*(torch.stack(x) for x in zip(*batch))))
```

Comment:
Might be an overkill to use a DataLoader here, but the intention is to use the transform the same way we will suggest to the users to use it.